service/history/workflowResetor_test.go

```diff
@@ -4371,3 +4371,165 @@ func (s *resetorSuite) TestResetWorkflowExecution_WithoutRunID() {
+
+	namespaceID := testNamespaceID
+
+	request := &historyservice.ResetWorkflowExecutionRequest{NamespaceId: namespaceID, ResetRequest: &workflowservice.ResetWorkflowExecutionRequest{}}
+
+	wid := "wId"
+	wfType := "wfType"
+	taskQueueName := "taskQueue"
+	forkRunID := uuid.New().String()
+	currRunID := uuid.New().String()
+
+	we := commonpb.WorkflowExecution{
+		WorkflowId: wid,
+		RunId:      forkRunID,
+	}
+
+	request.ResetRequest = &workflowservice.ResetWorkflowExecutionRequest{
+		Namespace:                 "testNamespace",
+		WorkflowExecution:         &we,
+		Reason:                    "test reset",
+		WorkflowTaskFinishEventId: 3,
+		RequestId:                 uuid.New().String(),
+	}
+
+	forkBranchToken := []byte("forkBranchToken")
+	taskQueue := &taskqueuepb.TaskQueue{
+		Name: taskQueueName,
+	}
+	// Prepare history event sequence.
+	readHistoryResponse := &persistence.ReadHistoryBranchByBatchResponse{
+		NextPageToken:    nil,
+		Size:             1000,
+		LastFirstEventID: int64(3),
+		History: []*historypb.History{
+			{
+				Events: []*historypb.HistoryEvent{
+					{
+						EventId:   1,
+						Version:   common.EmptyVersion,
+						EventType: enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_STARTED,
+						Attributes: &historypb.HistoryEvent_WorkflowExecutionStartedEventAttributes{WorkflowExecutionStartedEventAttributes: &historypb.WorkflowExecutionStartedEventAttributes{
+							WorkflowType: &commonpb.WorkflowType{
+								Name: wfType,
+							},
+							TaskQueue:                taskQueue,
+							Input:                    payloads.EncodeString("testInput"),
+							WorkflowExecutionTimeout: timestamp.DurationPtr(100 * time.Second),
+							WorkflowRunTimeout:       timestamp.DurationPtr(50 * time.Second),
+							WorkflowTaskTimeout:      timestamp.DurationPtr(200 * time.Second),
+						}},
+					},
+					{
+						EventId:   2,
+						Version:   common.EmptyVersion,
+						EventType: enumspb.EVENT_TYPE_WORKFLOW_TASK_SCHEDULED,
+						Attributes: &historypb.HistoryEvent_WorkflowTaskScheduledEventAttributes{WorkflowTaskScheduledEventAttributes: &historypb.WorkflowTaskScheduledEventAttributes{
+							TaskQueue:           taskQueue,
+							StartToCloseTimeout: timestamp.DurationPtr(100 * time.Second),
+						}},
+					},
+				},
+			},
+			{
+				Events: []*historypb.HistoryEvent{
+					{
+						EventId:    3,
+						Version:    common.EmptyVersion,
+						EventType:  enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_TIMED_OUT,
+						Attributes: &historypb.HistoryEvent_WorkflowTaskTimedOutEventAttributes{WorkflowTaskTimedOutEventAttributes: &historypb.WorkflowTaskTimedOutEventAttributes{}},
+					},
+				},
+			},
+		},
+	}
+
+	eid := int64(0)
+	eventTime := time.Unix(0, 1000).UTC()
+	for _, be := range readHistoryResponse.History {
+		for _, e := range be.Events {
+			eid++
+			if e.GetEventId() != eid {
+				s.Fail(fmt.Sprintf("inconintous eventID: %v, %v", eid, e.GetEventId()))
```

Comment:
[Nit] non-contiguous or non-continuous

Reply:
also, how could this happen?

Reply:
also curious when this would actually happen given we are the ones creating this mocked out data structure in the first place? We could have just assigned the eventTime variable inline for each event as well

Reply:
This can only happen if something is messed up in the reset logic in a way that creates gaps in event IDs. We do this check in other tests so it's there for consistency reasons.